### DISCUSSION TITLE
Log Telemetry

### DISCUSSION THREAD

**Discussion by shivchelwa** [open-telemetry/opentelemetry-java#3242] [Q&A] [answered]
The OpenTelemetry Specification discusses log correlation with trace and metrics. Please refer below link:
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/overview.md

Is there any extension available for log4j to enhance existing application log with span-id and trace-id?

I see some related work at https://github.com/open-telemetry/opentelemetry-java/tree/main/sdk-extensions/logging. But the documentation is not enough to understand if its consumable.

**Comment by jkwatson** [comment, marked as answer]
Hi @shivchelwa . Head over to https://github.com/open-telemetry/opentelemetry-java-instrumentation where there is log4j instrumentation that does this.
